services/apps/alcs/src/portal/notice-of-intent-document/notice-of-intent-document.controller.ts

```diff
@@ -0,0 +1,162 @@
+import { Mapper } from '@automapper/core';
+import { InjectMapper } from '@automapper/nestjs';
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Patch,
+  Post,
+  Req,
+  UseGuards,
+} from '@nestjs/common';
+import { ApiOAuth2 } from '@nestjs/swagger';
+import * as config from 'config';
+import { ApplicationDocumentDto } from '../../alcs/application/application-document/application-document.dto';
+import { VISIBILITY_FLAG } from '../../alcs/application/application-document/application-document.entity';
+import { NoticeOfIntentDocumentDto } from '../../alcs/notice-of-intent/notice-of-intent-document/notice-of-intent-document.dto';
+import { NoticeOfIntentDocument } from '../../alcs/notice-of-intent/notice-of-intent-document/notice-of-intent-document.entity';
+import { NoticeOfIntentDocumentService } from '../../alcs/notice-of-intent/notice-of-intent-document/notice-of-intent-document.service';
+import { NoticeOfIntentService } from '../../alcs/notice-of-intent/notice-of-intent.service';
+import { PortalAuthGuard } from '../../common/authorization/portal-auth-guard.service';
+import { DOCUMENT_TYPE } from '../../document/document-code.entity';
+import { DOCUMENT_SYSTEM } from '../../document/document.dto';
+import { DocumentService } from '../../document/document.service';
+import { NoticeOfIntentSubmissionService } from '../notice-of-intent-submission/notice-of-intent-submission.service';
+import {
+  AttachExternalDocumentDto,
+  PortalNoticeOfIntentDocumentUpdateDto,
+} from './notice-of-intent-document.dto';
+
+@ApiOAuth2(config.get<string[]>('KEYCLOAK.SCOPES'))
+@UseGuards(PortalAuthGuard)
+@Controller('notice-of-intent-document')
+export class NoticeOfIntentDocumentController {
+  constructor(
+    private noticeOfIntentDocumentService: NoticeOfIntentDocumentService,
+    private noticeOfIntentSubmissionService: NoticeOfIntentSubmissionService,
+    private noticeOfIntentService: NoticeOfIntentService,
+    private documentService: DocumentService,
+    @InjectMapper() private mapper: Mapper,
+  ) {}
+
+  @Get('/application/:fileNumber')
+  async listApplicantDocuments(
+    @Param('fileNumber') fileNumber: string,
+    @Param('documentType') documentType: DOCUMENT_TYPE | null,
+    @Req() req,
+  ): Promise<NoticeOfIntentDocumentDto[]> {
+    await this.noticeOfIntentSubmissionService.verifyAccessByFileId(
+      fileNumber,
+      req.user.entity,
+    );
+
+    const documents = await this.noticeOfIntentDocumentService.list(
+      fileNumber,
+      [VISIBILITY_FLAG.APPLICANT],
+    );
+    return this.mapPortalDocuments(documents);
+  }
+
+  @Get('/:uuid/open')
+  async open(@Param('uuid') fileUuid: string, @Req() req) {
+    const document = await this.noticeOfIntentDocumentService.get(fileUuid);
+
+    //TODO: How do we know which documents applicant can access?
```

Comment:
I wonder if this is still valid and if we have a separate ticket that captures this

Reply:
definitely still valid